devito/tools.py

```diff
@@ -56,9 +63,22 @@ def aligned(a, alignment=16):
     extra = alignment / a.itemsize
     buf = np.empty(a.size + extra, dtype=a.dtype)
     ofs = (-buf.ctypes.data % alignment) / a.itemsize
-    aa = buf[ofs:ofs+a.size].reshape(a.shape)
+    aa = buf[ofs:ofs + a.size].reshape(a.shape)
     np.copyto(aa, a)
 
     assert (aa.ctypes.data % alignment) == 0
 
     return aa
+
+
+def get_cache_line_size():
+    """Gets cache line size
+    :returns: cache line size"""
+    if "Linux" in platform.platform():
+        with open('/sys/devices/system/cpu/cpu0/cache/index0/coherency_line_size') as f:
+            return int(f.read())
+    else:  # Running on MAC
+        command = "/usr/sbin/sysctl -a | grep hw.cachelinesize"
```

Comment:
Would the `sysctl` trick not work on Linux as well? If not, can we get a comment in here that explains the special-casing? And also, have we confirmed that this stuff works on MIC?